src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1500,21 +1503,32 @@ def _get_azext_module(extension_name, module_name):
         raise CLIError(ie)
 
 
-def _install_dev_spaces_extension(extension_name):
+def _install_or_update_dev_spaces_extension(extension_name, update_extension):
```

Comment:
I do not understand the reason to combine the update and install in one method other than reuse the `try-except`, which I don't think to be a good reason. Using a boolean parameter to essentially make a method to play two completely different role is an anti-pattern.

Reply:
In invocation sequence here is therefore twisted and unclear.

Reply:
I was trying to re-use the code. But I can separate them so have more clarity. Reverting changes to install method.